### PR TITLE
Add deriving lib to CI.

### PR DESCRIPTION
**Pull request by arthuraa**
**Kind:** infrastructure.

Fixes / closes #????

- [ ] Added / updated test-suite

- [ ] Corresponding documentation was added / updated (including any warning and error messages added / removed / modified).
- [ ] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

### CODE REVIEW

**Comment by ejgallego** [comment]
Cool @arthuraa ! I think you need to add the `ci-deriving.sh` too.

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -820,6 +820,11 @@ library:ci-vst:
   - build:edge+flambda
   - library:ci-flocq
 
+library:ci-deriving:
+  extends: .ci-template-flambda
+  needs:
+  - library:ci-mathcomp
+
```
Comment (ejgallego) [inline .gitlab-ci.yml:827]: Seems need to be a stage later.

Comment (arthuraa) [reply .gitlab-ci.yml:828]: I put it in stage 3 now.

**Comment by arthuraa** [comment]
@ejgallego Let me know if there are any fixes that I can do on my end (I am assuming the errors I see are purely related to the CI).

**Comment by ejgallego** [comment]
> @ejgallego Let me know if there are any fixes that I can do on my end (I am assuming the errors I see are purely related to the CI).

@arthuraa I think the `install_ssreflect` command is not needed anymore, instead the dependency is declared at the level of the `.gitlab-ci.yml` file, and should be fine then. I am a bit puzzled by the failure in `ci-deriving` tho.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -820,6 +820,12 @@ library:ci-vst:
   - build:edge+flambda
   - library:ci-flocq
 
+library:ci-deriving:
+  extends: .ci-template-flambda
+  stage: stage-3
+  needs:
+  - library:ci-mathcomp
```
Comment (SkySkimmer) [inline .gitlab-ci.yml:827]:
```suggestion
  - build:edge+flambda
  - library:ci-mathcomp
```

Comment (ejgallego) [reply .gitlab-ci.yml:828]: Oh yes, the reason is that as of today gitlab's `needs:` don't pull transitive deps.

File: dev/ci/ci-deriving.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+ci_dir="$(dirname "$0")"
+. "${ci_dir}/ci-common.sh"
+
+install_ssreflect
```
Comment (ejgallego) [inline dev/ci/ci-deriving.sh:6]: This is not needed I think; @SkySkimmer , should we just remove these functions and the couple of users from CI? I'm gonna try a PR.

**Comment by ejgallego** [comment]
@arthuraa need to set the `x` bit on the ci-deriving script.

**Comment by arthuraa** [comment]
> @arthuraa need to set the `x` bit on the ci-deriving script.

Sorry, I don't understand what this means. I looked at the `ci-quickchick.sh` script as a reference, but I didn't find any `x` there. Should I look somewhere else?

**Comment by ejgallego** [comment]
Error seems unrelated.

@coqbot: merge now

**Review by ejgallego — APPROVED** [review]

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**